### ISSUE TITLE
Feature Request: local_exec parallelism controls

### ISSUE THREAD

**Issue by zpbonjour** [hashicorp/terraform#34931]
### Terraform Version

```shell
1.7.4
```

### Use Cases

I need to execute a CLI command several times per apply run based on some files in a local directory.  I do not want to create a local_exec null resource for each of these files and create a mess of "depends_on" logic when a for_each loop could work so much better.  The problem is that the CLI being executed by local_exec cannot be run more than one at a time as it locks the remote resource.  This causes all other parallel executions to fail.

### Attempted Solutions

Setting the global flag "-parallelism=1" solves my issue, but it slows down my entire apply run when I only need one specific resource block to not be executed in parallel.

### Proposal

```
locals {
  my_files = fileset(path.module, "*.json")
}

resource "null_resource" "example" {
  for_each = local.my_files
  provisioner "local-exec" {
    command =  "mycommand -file $MY_FILE"
    parallelism = False  # NEW Option
    environment = {
      MY_FILE = each.value
    }
  }
}
```
Or as part of the APPLY parameters
```
terraform apply -var-file=vars.tfvars -no-parallelism=null_resource.example
```

**Comment by jbardin** [comment]
Duplicate of #14258
